service/dps/storm/common/src/test/java/eu/europeana/cloud/service/dps/storm/spouts/kafka/MCSReaderSpoutTest.java

```diff
@@ -128,7 +129,12 @@ public void shouldEmitTheFilesWhenNoRevisionIsSpecified() throws Exception {
         when(representationIterator.hasNext()).thenReturn(true, false);
         when(representationIterator.next()).thenReturn(representation);
         when(fileServiceClient.getFileUri(eq(SOURCE + CLOUD_ID), eq(SOURCE + REPRESENTATION_NAME), eq(SOURCE + VERSION), eq("fileName"))).thenReturn(new URI(FILE_URL)).thenReturn(new URI(FILE_URL));
-        mcsReaderSpout.taskDownloader.execute(DATASET_URLS.name(), dpsTask);
+
+        ExecutorService executorService = Executors.newFixedThreadPool(MCSReaderSpout.INTERNAL_THREADS_NUMBER);
```

Comment:
I  understand what you are trying to do in here. However,  I would suggest to have a seperate test class TaskExecutor so you can test all those TaskExecutor functionalities. Currently MCSReaderSpout doesn't have those functionality anymore so why to test them in here!